python/lsst/ts/tcpip/utils.py

```diff
@@ -47,9 +50,12 @@ async def close_stream_writer(writer: asyncio.StreamWriter) -> None:
     """
     try:
         writer.close()
-        await writer.wait_closed()
+        # Work around https://bugs.python.org/issue39758
+        await asyncio.wait_for(writer.wait_closed(), timeout=5)
     except ConnectionResetError:
         pass
```

Comment:
Not part of the code changes, but please log a warning for this error too. It could help to diagnose connection problems between the ESS CSC and the ESS Controller, for instance.